vm/src/vm/runners/cairo_pie.rs

```diff
@@ -199,6 +231,41 @@ mod serde_impl {
         )
     }
 
+    impl CairoPieMemory {
+        pub fn to_bytes(&self) -> Vec<u8> {
+            // Missing segment and memory holes can be ignored
+            // as they can be inferred by the address on the prover side
+            let values = &self.0;
+            let mem_cap = values.len() * ADDR_BYTE_LEN + values.len() * FIELD_BYTE_LEN;
+            let mut res = Vec::with_capacity(mem_cap);
+
+            for ((segment, offset), value) in values.iter() {
+                let mem_addr = ADDR_BASE + *segment as u64 * OFFSET_BASE + *offset as u64;
+                res.extend_from_slice(mem_addr.to_le_bytes().as_ref());
+                match value {
+                    // Serializes RelocatableValue(little endian):
+                    // 1bit |   SEGMENT_BITS |   OFFSET_BITS
+                    // 1    |     segment    |   offset
```

Comment:
This is 64 bits, so `u64` seems like a better choice than `BigUint`.

Reply:
Its 8 bits for the address (which is built as an u64 in line 243) but in this case we are handling a memory value that is a relocatable, which takes 32 bytes

Reply:
`Relocatable` is 8 bytes AFAIR.